### PR TITLE
fix replay issue

### PR DESCRIPTION
**Pull request by gerben-stavenga**
### Description

There was a swap of checks creating different error codes because different error was checked first

### CODE REVIEW

**Review by runtian-zhou — APPROVED** [review]

**Comment by runtian-zhou** [comment]
Let's wait for at least the mainnet signal comes back before we land!

**Comment by runtian-zhou** [comment]
https://github.com/aptos-labs/aptos-core/actions/runs/4535567244/jobs/7991183765

Seems that mainnet has passed!

**Review by movekevin — APPROVED** [review]

**Pull request merged**
